src/test/ui/feature-gates/feature-gate-trait_upcasting.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0308]: mismatched types
+  --> $DIR/feature-gate-trait_upcasting.rs:11:25
+   |
+LL |     let bar: &dyn Bar = foo;
+   |                         ^^^ expected trait `Bar`, found trait `Foo`
+   |
+   = note: expected type `&dyn Bar`
+              found type `&dyn Foo`
+
+error: aborting due to previous error
```

Comment:
Before merging the PR, we should make sure that the gating actually suggests the feature gate so that people are invited to try it out.

Reply:
I think this will be rather non-trivial, but let me have a look. If so, it can always come in a follow-up PR.

Reply:
Sure. It needs to happen at some point tho, so just don't forget about it :)

Reply:
What probably makes sense is doing a "is supertrait" query on this error path. Then we can suggest it only in appropriate cases I think.